AssemblyAutoBoard/AssemblyAutoBoard.ino

```diff
@@ -263,228 +256,260 @@ void loop() {
         Serial.println("Crimp Cycle Activated");
-        digitalWrite(PanelLed4, HIGH);
-        digitalWrite(SolenoidArray[4], HIGH);
+        digitalWrite(panelLed4, HIGH);
+        digitalWrite(solenoidArray[4], HIGH);
         previousTimer4 = currentTime;
-        CrimpNext = 1;
+        crimpNext = 1;
       }
-      if ((CrimpNext == 1) && (currentTime - previousTimer4 >= sysArray[3])) {
+      if ((crimpNext == 1) && (currentTime - previousTimer4 >= sysArray[3])) {
         previousTimer4 = currentTime;
-        digitalWrite(SolenoidArray[5], HIGH);
+        digitalWrite(solenoidArray[5], HIGH);
         Serial.println("Crimp Cycle | Crimp");
-        CrimpNext = 2;
+        crimpNext = 2;
       }
-      if ((CrimpNext == 2) && (currentTime - previousTimer4 >= sysArray[4])) {
+      if ((crimpNext == 2) && (currentTime - previousTimer4 >= sysArray[4])) {
         Serial.println("Crimp Cycle | Reset");
         previousTimer4 = currentTime;
-        digitalWrite(SolenoidArray[5], LOW);
-        digitalWrite(SolenoidArray[4], LOW);
-        digitalWrite(PanelLed4, LOW);
-        CrimpNext = 0;
+        digitalWrite(solenoidArray[5], LOW);
+        digitalWrite(solenoidArray[4], LOW);
+        digitalWrite(panelLed4, LOW);
+        crimpNext = 0;
       }
       // Hook Cycle
-      HookLoop = digitalRead(SensorArray[2]);
-      if ((HookLoop == LOW) && (HookNext == 0)) {
-        Serial.println("Hook Cycle Activated");
-        digitalWrite(PanelLed2, HIGH);
-        boolean HookCheck;
-        HookCheck = digitalRead(SensorArray[0]);
-        if (HookCheck == HIGH) {
-          Serial.println("ERROR: Hook Check failed");
-          lcd.setCursor(0, 3);
-          lcd.print("ERROR: Hook Check");
-          preLCDClear = currentTime;
-          //Error = 1;
-          digitalWrite(PanelLed2, LOW);
-          FeedLoop = 0;
-          FeedNext = 0;
-          digitalWrite(ErrorLed, HIGH);
+      hookLoop = digitalRead(sensorArray[2]);
+      if ((hookLoop == LOW) && (hookNext == 0)) {
+        if (currentTime - previousTimer3 >= sysArray[6]) {
+          Serial.println("Hook Cycle Activated");
+          digitalWrite(panelLed2, HIGH);
+          boolean hookCheck;
+          hookCheck = digitalRead(sensorArray[0]);
+          if (hookCheck == HIGH) {
+            Serial.println("ERROR: Hook Check failed");
+            lcd.setCursor(0, 3);
+            lcd.print("ERROR: Hook Check");
+            preLCDClear = currentTime;
+            //partError = 1;
+            digitalWrite(panelLed2, LOW);
+            feedLoop = 0;
+            feedNext = 0;
+            digitalWrite(errorLed, HIGH);
+          }
+          if (hookCheck == LOW) {
+            previousTimer3 = currentTime;
+            digitalWrite(solenoidArray[1], HIGH);
+            hookNext = 1;
+          }
         }
-        if (HookCheck == LOW) {
+        if ((currentTime - previousTimer3 < sysArray[6]) && (currentTime - previousTimer3 >= buttonWait) && (mpsEnable >= 4)) {
```

Comment:
Should consider using a separate changeable variable for the sensor ignore period.